state/in-memory/in_memory_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inmemory
+
+import (
+	"testing"
+
+	"github.com/agrea/ptr"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dapr/kit/logger"
+
+	"github.com/dapr/components-contrib/state"
+)
+
+func TestReadAndWrite(t *testing.T) {
+	ctl := gomock.NewController(t)
+
+	defer ctl.Finish()
+
+	store := NewInMemoryStateStore(logger.NewLogger("test"))
+	store.Init(state.Metadata{})
+
+	t.Run("test set 1", func(t *testing.T) {
```

Comment:
Please rename it as happy or bad cases and cover them.

Reply:
fixed